src/app/weather-forecast/weatherForecastApiService/weather-forecast-api.service.ts

```diff
@@ -0,0 +1,17 @@
+
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class WeatherForecastApiService {
+
+    constructor(private http: HttpClient) { }
+
+    private wfApiUrl = 'https://api.openweathermap.org/data/2.5/weather?q=Pisa,it&appid=d5754a8efdf1ead276cffa4f8250f1e1';
```

Comment:
You can move this inside the `getObs`function. Try to extract `apiid` to make easier further changes